configure.ac

```diff
@@ -300,51 +299,63 @@ AS_IF([test -z $enable_checks && test "x$DISABLE_CHECK" = "xprobably"], [
   enable_checks=no
 ])
 
-CHECK_LEVEL="0"
+BOUT_CHECK_LEVEL=0
 AS_IF([test "x$enable_checks" != "xno" && test "x$enable_checks" != "x0"], [
   AC_MSG_NOTICE([Run-time checking enabled])
   AS_CASE([$enable_checks],
     [1], [AC_MSG_NOTICE([ -> Level 1 (Basic checking)])
           CXXFLAGS="$CXXFLAGS -DCHECK=1"
-          CHECK_LEVEL="1"],
+          BOUT_CHECK_LEVEL=1],
     [3], [AC_MSG_NOTICE([ -> Level 3 (Full checking + stack tracing)])
           CXXFLAGS="$CXXFLAGS -DCHECK=3"
-          CHECK_LEVEL="3"],
+          BOUT_CHECK_LEVEL=3],
     [AC_MSG_NOTICE([ -> Level 2 (Basic checking + stack tracing)])
      CXXFLAGS="$CXXFLAGS -DCHECK=2"
-     CHECK_LEVEL="2"])
+     BOUT_CHECK_LEVEL=2])
 ], [
   AC_MSG_NOTICE([Run-time checking disabled])
 ])
 
+BOUT_USE_SIGNAL=no
 AS_IF([test "x$enable_signal" != "xno"], [
   AC_MSG_NOTICE([Segmentation fault handling enabled])
-  CXXFLAGS="$CXXFLAGS -DSIGHANDLE"
+  BOUT_USE_SIGNAL=yes
 ], [
   AC_MSG_NOTICE([Segmentation fault handling disabled])
 ])
 
+BOUT_USE_COLOR=no
 AS_IF([test "x$enable_color" != "xno"], [
   AC_MSG_NOTICE([Output coloring enabled])
-  CXXFLAGS="$CXXFLAGS -DLOGCOLOR"
+  BOUT_USE_COLOR=yes
 ], [
   AC_MSG_NOTICE([Output coloring disabled])
 ])
 
+BOUT_USE_TRACK=no
 AS_IF([test "x$enable_track" = "xyes"], [
   AC_MSG_NOTICE([Field name tracking enabled])
-  CXXFLAGS="$CXXFLAGS -DTRACK"
+  BOUT_USE_TRACK=yes
 ], [
   AC_MSG_NOTICE([Field name tracking disabled])
 ])
 
+BOUT_USE_SIGFPE=no
 AS_IF([test "x$enable_sigfpe" = "xyes"], [
   AC_MSG_NOTICE([Signaling floating point exceptions enabled])
-  CXXFLAGS="$CXXFLAGS -DBOUT_FPE"
+  BOUT_USE_SIGFPE=yes
 ], [
   AC_MSG_NOTICE([Signaling floating point exceptions disabled])
 ])
 
+BOUT_USE_OUTPUT_DEBUG=no
+AS_IF([test "x$enable_output_debug" = "xyes"], [
```

Comment:
Might be nice to enable if we have `BOUT_CHECK_LEVEL=3`?

Reply:
Yes, that seems wise

Reply:
It's now forced on at CHECK=3, ditto for CMake. Still requires `-v -v` to enable at runtime though